workspace/viewer/PdfJsViewer.ts

```diff
@@ -692,15 +719,29 @@ export class PdfJsViewer {
 
   scrollUpdate() {
     const visible = this.getCurrentVisiblePages();
-    const numVisible = visible.views.length;
+    if(visible){
+      this.currentPage = visible.first.id;
+      for (const page of visible.views) {
+        const pageNumber = page.view.id;
+        this.renderPage(pageNumber);
+      }
+
+      //Trying to render next or prev page
+      if (this.scroll.down) {
+        let nextPageIndex = visible.last.id;
+        // ID's start at 1 so no need to add 1.
+        if (this.loadedPages[nextPageIndex] &&
+          this.loadedPages[nextPageIndex].renderingState !== RenderingStates.FINISHED) {
```

Comment:
It seems that you can just pass the page to `renderPage` method because it will check the state by itself. By the way, any reason not to pass the page object instead of page index to `renderPage`?

Reply:
Done. It seems much prettier to use an object.